src/code_snippets/remove_repeated_characters.go.tmpl

```diff
@@ -0,0 +1,46 @@
+{{/*
+    Removes the repeated letters in a text like 'byyeeee' => 'bye'
+    Useful to add it in another more complex commands.
+
+    Author: Faculord <https://github.com/LattandiFacundo>
+*/}}
+
+{{/*Configuration variables*/}}
+{{ $toFix := "ooooooppppppppppppppppppppsssssssssssssssssss"}}
+{{ $exceptions := dict
+"o" 3
+"p" 2
+}}
+{{ $defaultThreshold := 1 }}
+{{/*End of configuration variables*/}}
+
+{{ $splitted := split $toFix "" }}
+{{ $before := "" }} {{ $fixed := "" }} {{ $remaining := 0 }} {{ $catchedException := false }}
+
+{{ range $e := $splitted }}
+	{{ range $exception, $ignoreMe := $exceptions}}
+		{{- if and (ne $before $e) (eq $e $exception) }}
+			{{- $remaining = (toInt ($exceptions.Get $e)) }}
+			{{- $fixed = print $fixed $e }}
+			{{- $catchedException = true }}
+		{{- else if and (eq $e $exception) (ge $remaining 1) }}
+			{{- $fixed = print $fixed $e }}
+			{{- $catchedException = true }}
+		{{- else if ne $before $e}}
+			{{- $catchedException = false }}
+		{{- end }}
+	{{- end }}
```

Comment:
Can you explain what this does, and why `.Get` isn't sufficient?

Reply:
The firts one is for the first catch of the exception, in the second one when it print but do not actualize the remaining and the other one to let the default threshold work when there is no exception after exeptions range.

Reply:
That still doesn't quite answer why `.Get` isn't sufficient. The reason I'm concerned about this is that `.Get` is faster than a loop through the map (which could cause a slowdown with large input + exceptions). Also, it's definitely possible using `.Get` (see https://github.com/yagpdb-cc/yagpdb-cc/pull/238#issuecomment-925977801) and likely also cleaner.